src/components/TableSettings/TableSettings.tsx

```diff
@@ -39,16 +46,23 @@ export interface TableSettingsProps<TData> extends TableSettingsOptions {
 
 const POPUP_PLACEMENT: PopupPlacement = ['bottom-end', 'bottom', 'top-end', 'top', 'left', 'right'];
 
+const NESTED_COLUMNS_PREFIX = 'nested_columns_count_';
+
 export const TableSettings = <TData extends unknown>({
     table,
     sortable = true,
     filterable = true,
+    enableSearch = false,
+    searchPlaceholder = '',
     onSettingsApply,
 }: TableSettingsProps<TData>) => {
     const anchorRef = React.useRef<HTMLButtonElement>(null);
     const [open, setOpen] = React.useState<boolean>(false);
-    const columns = table.getAllColumns();
-    const filteredColumns = React.useMemo(() => columns.filter(isDisplayedColumn), [columns]);
+    const [search, setSearch] = React.useState('');
+    const [expandNestedColumns, setExpandNestedColumns] = React.useState(false);
+    const allColumns = table.getAllColumns();
```

Comment:
Are the columns needed on every render? Why is this not inside the memo?

Reply:
[fixed](https://github.com/gravity-ui/table/pull/107/commits/8fc67d9320db1ef19ebb3570cb7a1fe6aa857aac#diff-a50b5646081d6956f39653b4a41fb1ff1396e2829b79af86ff189e7a173f666eR63)